module/os/linux/zfs/zfs_vnops_os.c

```diff
@@ -3517,6 +3531,16 @@ zfs_putpage(struct inode *ip, struct page *pp, struct writeback_control *wbc)
 		zfs_rangelock_exit(lr);
 
 		if (wbc->sync_mode != WB_SYNC_NONE) {
+			/*
+			 * Speed up any non-sync page writebacks since
+			 * they may take several seconds to complete.
+			 * Refer to the comment in zpl_fsync() (when
+			 * HAVE_FSYNC_RANGE is defined) for details.
+			 */
+			if (atomic_load_32(&zp->z_async_writes_cnt) > 0) {
+				zil_commit(zfsvfs->z_log, zp->z_id);
```

Comment:
My oversight, we do want to keep the `ZPL_*` calls, but only around this `zil_commit` and not the atomics.  Can you please add them back to this conditional block, sorry about the hassle.

```c
		if (atomic_load_32(&zp->z_async_writes_cnt) > 0) {
			ZPL_ENTER(zfsvfs);
			ZPL_VERIFY_ZP(zp);
			zil_commit(zfsvfs->z_log, zp->z_id);
			ZPL_EXIT(zfsvfs);
		}						
```

Reply:
no worries, yes, i'll add them back again in a moment

Reply:
pushed